serverless-plugin/alarms-dynamodb.js

```diff
@@ -0,0 +1,78 @@
+'use strict'
+
+const { makeResourceName } = require('./util')
+
+/**
+ * @param {object} dynamoDbAlarmConfig The fully resolved alarm configuration
+ */
+module.exports = function DynamoDbAlarms (dynamoDbAlarmConfig, context) {
+  return {
+    createDynamoDbAlarms
+  }
+
+  /**
+   * Add all required DynamoDB alarms to the provided CloudFormation template
+   * based on the tables and their global secondary indices.
+   *
+   * @param {CloudFormationTemplate} cfTemplate A CloudFormation template object
+   */
+  function createDynamoDbAlarms (cfTemplate) {
+    const tableResources = cfTemplate.getResourcesByType(
+      'AWS::DynamoDB::Table'
+    )
+
+    for (const [tableResourceName, tableResource] of Object.entries(tableResources)) {
+      const tableName = tableResource.Properties.TableName
+      const tableDimensions = [{ Name: 'TableName', Value: tableName }]
+
+      const alarms = [
+        createAlarm(tableName, tableDimensions, 'ReadThrottleEvents', makeResourceName('Table', `${tableName}`, 'ReadThrottleEvents')),
+        createAlarm(tableName, tableDimensions, 'WriteThrottleEvents', makeResourceName('Table', `${tableName}`, 'WriteThrottleEvents')),
+        createAlarm(tableName, tableDimensions, 'UserErrors', makeResourceName('Table', `${tableName}`, 'UserErrors')),
+        createAlarm(tableName, tableDimensions, 'SystemErrors', makeResourceName('Table', `${tableName}`, 'SystemErrors'))
+      ]
+
+      for (const gsi of tableResource.Properties.GlobalSecondaryIndexes || []) {
+        const gsiName = gsi.IndexName
+        const gsiDimensions = [...tableDimensions, { Name: 'GlobalSecondaryIndex', Value: gsiName }]
+        const identifier = `${tableName}${gsiName}`
+        alarms.push(...[
+          createAlarm(identifier, gsiDimensions, 'ReadThrottleEvents', makeResourceName('GSI', `${tableResourceName}${gsiName}`, 'ReadThrottleEvents')),
+          createAlarm(identifier, gsiDimensions, 'WriteThrottleEvents', makeResourceName('GSI', `${tableResourceName}${gsiName}`, 'WriteThrottleEvents'))
+        ])
+      }
```

Comment:
```suggestion
        alarms.push(
          createAlarm(identifier, gsiDimensions, 'ReadThrottleEvents', makeResourceName('GSI', `${tableResourceName}${gsiName}`, 'ReadThrottleEvents')),
          createAlarm(identifier, gsiDimensions, 'WriteThrottleEvents', makeResourceName('GSI', `${tableResourceName}${gsiName}`, 'WriteThrottleEvents'))
        )
      }
```

This should work. So we can avoid constructing an array just to use the spread operator